### ISSUE TITLE
AMQP : support "confirm" Sink and Flow producers

### ISSUE THREAD

**Issue by jcare44** [akka/alpakka#1132]
Hi,
I think it would be great for the AMQP lib to handle publish confirms, both in a specific Sink and via a Flow. I like [the approach of the alpakka Kafka lib on that](https://doc.akka.io/api/akka-stream-kafka/0.22/akka/kafka/scaladsl/Producer$.html), and I think it would be great to implement something similar for AMQP.
What do you think of it ?

**Comment by juanjoDiaz** [comment]
Can you elaborate a bit on what you are suggesting?

Do you mean having a passthrough materialized view? (I found myself needing that recently).

**Comment by 2m** [comment]
Yes, I aggree that it would be useful to have AMQP producer modeled as a Flow, where the input to the Flow would be `OutgoingMessage` and the flow would emit confirmations that the publish has been successful or not.
